### ISSUE TITLE
Description of extra_fields

### ISSUE THREAD

**Issue by schwehr** [stac-utils/pystac#318]
Another argument for working towards more type annotations.

The doc strings say `dict or None`, but in catalog.py:

```python
    def __init__(self,
                 id,
                 description,
                 title=None,
                 stac_extensions=None,
                 extra_fields=None,
                 href=None,
                 catalog_type=CatalogType.ABSOLUTE_PUBLISHED):
        super().__init__(stac_extensions)

        self.id = id
        self.description = description
        self.title = title
        if extra_fields is None:
            self.extra_fields = {}  # <--- This isn't None
        else:
            self.extra_fields = extra_fields
```

I think these should just say `Dict`.

`catalog.py` says:

```
    Attributes:
        id (str): Identifier for the catalog.
        description (str): Detailed multi-line description to fully explain the catalog.
        title (str or None): Optional short descriptive one-line title for the catalog.
        stac_extensions (List[str] or None): Optional list of extensions the Catalog implements.
        extra_fields (dict or None): Extra fields that are part of the top-level JSON properties
            of the Catalog.
        links (List[Link]): A list of :class:`~pystac.Link` objects representing
            all links associated with this Catalog.
        catalog_type (str or None): The catalog type, or None if not known.
```

`collections.py` says:

```
    Attributes:
        id (str): Identifier for the collection.
        description (str): Detailed multi-line description to fully explain the collection.
        extent (Extent): Spatial and temporal extents that describe the bounds of
            all items contained within this Collection.
        title (str or None): Optional short descriptive one-line title for the collection.
        stac_extensions (List[str]): Optional list of extensions the Collection implements.
        keywords (List[str] or None): Optional list of keywords describing the collection.
        providers (List[Provider] or None): Optional list of providers of this Collection.
        properties (dict or None): Optional dict of common fields across referenced items.
        summaries (dict or None): An optional map of property summaries,
            either a set of values or statistics such as a range.
        links (List[Link]): A list of :class:`~pystac.Link` objects representing
            all links associated with this Collection.
        extra_fields (dict or None): Extra fields that are part of the top-level JSON properties
            of the Catalog.
```

**Comment by lossyrob** [comment]
Have you seen #309? This places full type annotations into the library. Btw, that just became ready for review, so thoughts are appreciated, though there are known issues with docstrings.

I think now that there are type annotations (once that pr is merged) I'd recommend removing type information from the docstrings. It would be great to get help with this.

**Comment by schwehr** [comment]
Cool.  I'll take a look.  Happy to help.  I'm still just a beginner at type annotations, but I've been doing more of them lately.

**Comment by lossyrob** [comment]
Will be solved by #323
